blas/testblas/zaxpy.go

```diff
@@ -0,0 +1,119 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testblas
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+)
+
+type Zaxpyer interface {
+	Zaxpy(n int, alpha complex128, x []complex128, incX int, y []complex128, incY int)
+}
+
+func ZaxpyTest(t *testing.T, impl Zaxpyer) {
+	rnd := rand.New(rand.NewSource(1))
+
+	for tc, test := range []struct {
+		alpha complex128
+		x, y  []complex128
+
+		want    []complex128
+		wantRev []complex128
+	}{
+		{
+			alpha:   0,
+			x:       []complex128{2 + 3i, 3 + 4i, 4 + 5i},
+			y:       []complex128{20 + 30i, 30 + 40i, 40 + 50i},
+			want:    []complex128{20 + 30i, 30 + 40i, 40 + 50i},
+			wantRev: []complex128{20 + 30i, 30 + 40i, 40 + 50i},
+		},
+		{
+			alpha:   1,
+			x:       []complex128{2 + 3i, 3 + 4i, 4 + 5i},
+			y:       []complex128{20 + 30i, 30 + 40i, 40 + 50i},
+			want:    []complex128{22 + 33i, 33 + 44i, 44 + 55i},
+			wantRev: []complex128{24 + 35i, 33 + 44i, 42 + 53i},
+		},
+		{
+			alpha:   3 + 7i,
+			x:       []complex128{2 + 3i, 3 + 4i, 4 + 5i},
+			y:       []complex128{20 + 30i, 30 + 40i, 40 + 50i},
+			want:    []complex128{5 + 53i, 11 + 73i, 17 + 93i},
+			wantRev: []complex128{-3 + 73i, 11 + 73i, 25 + 73i},
+		},
```

Comment:
The main loop in `asm/c128` is 4-wide and the future AVX version will be 8-wide, so there should be test entries of length 10+ in here.